### ISSUE TITLE
The "Move symbol to ..." refactoring displaces trailing comments  in the destination module (including `type: ignore`, etc.)

### ISSUE THREAD

**Issue by blakeNaccarato** [microsoft/pylance-release#4136]
## Overview

The "Move symbol to ..." refactoring displaces trailing comments in the destination module. This is not so bad for whole-line comments, but it also "steals" in-line comments like `type: ignore` or `noqa...` from the last code line in the file.

## Environment data

-   Language Server version: 2023.3.30 (pyright 68a8e98a)
-   OS and version: Windows 10 Pro 22H2
-   Python version: 3.11.2

## Code Snippet

This is reproducible by having the following top-level modules in the working package, say `src/myproject` and having Pylance recognize this package as being able to be refactored with the "Move symbol to ..." refactoring. See the reproduction below.

`src/myproject/source.py`

```Python
b = 2
<newline at end of file>
```

`src/myproject/destination_with_trailing_comment.py`

```Python
a = 1  # type: ignore
<newline at end of file>
```

## Repro Steps

1. Set `"python.analysis.refactoring.allowMovingSymbols": true,` in `settings.json`
2. Try to move a symbol to a file with a trailing in-line comment
3. The moved symbol will "steal" the trailing in-line comment from its original target

See also in the behaviors below some additional unnecessary newlines, but these are easily managed by formatting on save. The actual relevant structural bug is type-checking/linter-relevant comments are displaced.

## Expected behavior

Ideally the resulting files would look like the following. If the underlying implementation for moving symbols is fundamentally incapable of preserving trailing in-line comments (tokenizer/syntax tree funny business, etc.), perhaps a toast/notification should at least be raised in the VSCode interface that says something like "Trailing in-line or full-line comment in destination module may have been displaced" if the destination file has a trailing comment detected by regex.

`source.py`

```Python
<newline at end of file>
```

`destination_with_trailing_comment.py`

```Python
a = 1  # type: ignore
<blank line>
b = 2
<newline at end of file>
```

## Actual behavior

`source.py`

```Python
<blank line>
<newline at end of file>
```

`destination_with_trailing_comment.py`

```Python
a = 1
<two blank lines>
b = 2  # type: ignore
<newline at end of file>
```

## Logs

Will provide logs upon request, but I don't think it will be necessary.

**Comment by blakeNaccarato** [comment]
## Original

`source.py`

`newfile.py`

```Python
```

**Comment by heejaechang** [comment]
thank you for reporting the issue. working on the fix!

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.4.11, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023411-5-April-2023-PreRelease)

It should be noted that it places two blank lines still (that's on purpose at the moment). #4202 was logged to address this.
